File: frontend/src/lib/permissions.ts
import type { Role } from '../types'

export function canEditFloor(role: Role): boolean {
  return role === 'OWNER' || role === 'MANAGER'
}

export function canManageUsers(role: Role): boolean {
  return role === 'OWNER'
}

export function canSeatGuests(_role: Role): boolean {
  return true
}

export function canChangeStatus(_role: Role): boolean {
  return true
}

File: backend/app/core/permissions.py
from typing import Literal

Role = Literal["OWNER", "MANAGER", "HOST", "WAITER"]


def normalize_role(role: str) -> str:
    return role.strip().upper()


def can_edit_floor(role: str) -> bool:
    return normalize_role(role) in ("OWNER", "MANAGER")


def can_manage_users(role: str) -> bool:
    return normalize_role(role) == "OWNER"


def can_manage_menu(role: str) -> bool:
    return normalize_role(role) in ("OWNER", "MANAGER")


def can_manage_reservations(role: str) -> bool:
    return normalize_role(role) in ("OWNER", "MANAGER", "HOST")
